woocommerce/woocommerce-ios
# Search products by SKU: storage layer - add `filterKey` to `ProductSearchResults` entity

Part of #3157


- [x] ⚠️ Update release version in `MIGRATIONS.md` and check if the schema has changed when the subsequent PRs are approved ⚠️ 

### Description


To support searching products by SKU, we need to introduce a segmented control for the user to manually select whether they want to search by SKU or not due to performance reason (more details in p91TBi-96v-p2#comment-10714). In order to differentiate results in the two tabs (all products or products matched by SKU), we need to store `ProductSearchResults` for each filter tab. This PR only contains the storage layer changes for easier review. The diffs on the new model version 73 are in https://github.com/woocommerce/woocommerce-ios/commit/797b6d1ea9d1fe530f721378357a370295665495. Some notes:

- Due to the timing of the HACK Week right before the meetup, I'll update the migration notes with the correct release version
- This feature was already implemented in Android https://github.com/woocommerce/woocommerce-android/pull/6838
- If the schema is updated by the team before this is merged, I'll re-do the schema change

### Testing instructions


Please feel free to launch the app from another branch like `trunk` to make sure the migration works as expected.


---
